src/NServiceBus.Core/Pipeline/Incoming/IHandlingMetricsFactory.cs

```diff
@@ -0,0 +1,17 @@
+﻿namespace NServiceBus;
+
+using Pipeline;
+
+/// <summary>
+/// Factory for IHandlingMetrics needed to record metrics for the handler invocation.
+/// </summary>
+interface IHandlingMetricsFactory
```

Comment:
To be more explicit about what metric we’re referring to here, I’d rename this type to `IMessageHandlingMetricsFactory`